drivers/platform/stm32/stm32_gpio.c

```diff
@@ -97,6 +97,14 @@ static int32_t _gpio_init(struct gpio_desc *desc,
 	else
 		return -EINVAL;
 
+	switch (pextra->mode & GPIO_MODE) {
```

Comment:
Is this check really required? Since we have multiple `pextra` members that are directly assigned to the `extra` struct without any prior checks. (although it's good to have this kind of checks for all init params if possible)

Reply:
On stm32, (pextra->mode & GPIO_MODE) can have the following values:
```
#define MODE_INPUT                              (0x0UL << GPIO_MODE_Pos)
#define MODE_OUTPUT                             (0x1UL << GPIO_MODE_Pos)
#define MODE_AF                                 (0x2UL << GPIO_MODE_Pos)
#define MODE_ANALOG                             (0x3UL << GPIO_MODE_Pos)
```
It doesn't make sense for the GPIO driver to allow anything else than input and output. AF is alternate function (like SPI, I2C etc,) and ANALOG is for ADC/DAC pins.

This is why I put the check, I think it makes sense to have it there.